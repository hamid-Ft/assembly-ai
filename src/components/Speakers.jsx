/* eslint-disable react/prop-types */
const Speakers = ({ transcript }) => {
  return (
    <div>
      <h3>Speakers: </h3>
      <div>
        {transcript.utterances.map((utterance) => (
          <div key={utterance.start}>
            <span style={{ fontWeight: "bolder", paddingRight: "8px" }}>
              {utterance.speaker}:
            </span>
            <span>{utterance.text}</span>
          </div>
        ))}
      </div>
    </div>
  );
};

export default Speakers;
