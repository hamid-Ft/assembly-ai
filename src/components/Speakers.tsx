import { Transcript } from '../types/transcript';

const Speakers = ({ transcript }: { transcript: Transcript }) => {
    return (
        <div>
            <h3>Speakers: </h3>
            <div>
                {transcript.utterances?.map(utterance => (
                    <div key={utterance.start}>
                        <span className="font-bolder pr-2">
                            {`${utterance.speaker}: ${utterance.text}`}
                        </span>
                    </div>
                ))}
            </div>
        </div>
    );
};

export default Speakers;
