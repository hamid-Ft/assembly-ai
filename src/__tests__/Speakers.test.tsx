import { render, screen } from '@testing-library/react';
import Speakers from '../components/Speakers';
import '@testing-library/jest-dom';

test('renders Speakers component', async () => {
    const mockTranscript = {
        id: '001',
        utterances: [
            {
                start: 0,
                speaker: 'A',
                text: 'Hello',
                confidence: 0.86,
                end: 1.2234,
                words: [
                    {
                        text: 'Hello',
                        start: 0,
                        end: 1.2234,
                        confidence: 0.86,
                        speaker: 'A',
                    },
                ],
            },
            {
                start: 1.3483,
                speaker: 'B',
                text: 'Hi',
                confidence: 0.82,
                end: 2.2489,
                words: [
                    {
                        text: 'Hi',
                        start: 1.3483,
                        end: 2.2489,
                        confidence: 0.82,
                        speaker: 'B',
                    },
                ],
            },
        ],
    };
    render(<Speakers transcript={mockTranscript} />);

    const speaker1Element = await screen.findByText(/A: Hello/i);
    const speaker2Element = await screen.findByText(/B: Hi/i);

    expect(speaker1Element).toBeInTheDocument();
    expect(speaker2Element).toBeInTheDocument();
});

test('matches snapshot', () => {
    const mockTranscript = {
        id: '001',
        utterances: [
            {
                start: 0,
                speaker: 'A',
                text: 'Hello',
                confidence: 0.86,
                end: 1.2234,
                words: [
                    {
                        text: 'Hello',
                        start: 0,
                        end: 1.2234,
                        confidence: 0.86,
                        speaker: 'A',
                    },
                ],
            },
            {
                start: 1.3483,
                speaker: 'B',
                text: 'Hi',
                confidence: 0.82,
                end: 2.2489,
                words: [
                    {
                        text: 'Hi',
                        start: 1.3483,
                        end: 2.2489,
                        confidence: 0.82,
                        speaker: 'B',
                    },
                ],
            },
        ],
    };
    const { asFragment } = render(<Speakers transcript={mockTranscript} />);
    expect(asFragment()).toMatchSnapshot();
});
