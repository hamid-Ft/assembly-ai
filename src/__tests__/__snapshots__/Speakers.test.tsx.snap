// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
      Speakers: 
    </h3>
    <div>
      <div>
        <span
          class="font-bolder pr-2"
        >
          A: Hello
        </span>
      </div>
      <div>
        <span
          class="font-bolder pr-2"
        >
          B: Hi
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
